tracer/src/Datadog.Trace/DataStreamsMonitoring/DataStreamsContextPropagator.cs

```diff
@@ -31,7 +34,14 @@ public void Inject<TCarrier>(PathwayContext context, TCarrier headers)
     {
         if (headers is null) { ThrowHelper.ThrowArgumentNullException(nameof(headers)); }
 
-        headers.Add(DataStreamsPropagationHeaders.PropagationKey, PathwayContextEncoder.Encode(context));
+        var encodedBytes = PathwayContextEncoder.Encode(context);
+        var base64EncodedContext = Convert.ToBase64String(encodedBytes);
+        headers.Add(DataStreamsPropagationHeaders.PropagationKeyBase64, Encoding.UTF8.GetBytes(base64EncodedContext));
```

Comment:
In newer versions of .NET there's a method that converts from bytes into UTF-8 encoded base-64 bytes while skipping the intermediate string: `Base64.EncodeToUtf8` ([docs](https://learn.microsoft.com/en-us/dotnet/api/system.buffers.text.base64.encodetoutf8)). We have the code [in our repo](https://github.com/DataDog/dd-trace-dotnet/blob/bd582eb71638575c00fe904ee1b3a0a563de77d0/tracer/src/Datadog.Trace/Vendors/System.Memory/Buffers/Text/Base64.cs#L578) so we can use it from any version of .NET.

It will look something like:
```csharp
var encodedBytes = PathwayContextEncoder.Encode(context);

int base64Length = ((encodedBytes.Length + 2) / 3) * 4;
byte[] base64EncodedContextBytes = new byte[base64Length];
var status = Base64.EncodeToUtf8(encodedBytes , base64EncodedContextBytes, out _, out _);
// TODO: check that status == OperationStatus.Done

headers.Add(DataStreamsPropagationHeaders.PropagationKeyBase64, base64EncodedContextBytes);
```